### ISSUE TITLE
Discuss The Archiving of jedisct1/dnscrypt-proxy

### ISSUE THREAD

**Issue by snorkasaurus** [snorkasaurus/dnscrypt-proxy#7]
Today I noticed that the jedisct1/dnscrypt-proxy repository has been archived, which marks it as read-only. Back in November jedisct1 opened [this issue](https://github.com/jedisct1/dnscrypt-proxy/issues/769) saying that the repo would soon be archived, though I was unaware of the time what that actually meant. Now another month has passed and the repo is read-only.

Though some people have said that they think dnscrypt-proxy could continue to serve its purpose without any code changes, the read-only status of the repository also means that there can no longer be changes to the public resolver list. If people are going to continue to use dnscrypt-proxy we will at least need a new location to store and maintain the public resolver list (which I believe is also used by some other applications).

I am creating this issue here to discuss the future of the project. I have no real preference concerning which fork is used... just that the project continues to be functional.

I imagine there would also need to be some discussion about the future of the dnscrypt.org domain name and web site.  The site does not seem to be updated very frequently and I would probably be willing to take over management of it.  Would anyone consider that a conflict of interest since I am running the dnscrypt.ca resolvers?

Comments?

**Comment by Fusl** [comment]
> the read-only status of the repository also means that there can no longer be changes to the public resolver list.

A [new repository was created](https://github.com/DNSCrypt/dnscrypt-resolvers) to serve for just this purpose.

**Comment by snorkasaurus** [comment]
Excellent, thanks for the quick reply!

It does kind of leave me with some questions still.  Such as whether jedisct1 will continue to be the maintainer of the public resolver list or whether there will be any continued work on dnscrypt-proxy itself. Any thoughts?

**Comment by Fusl** [comment]
I don't have much information on the questions you asked but jedisct1 only recently created the [DNSCrypt GitHub team](https://github.com/DNSCrypt) silently adding 8 members (including me) to the team so I assume that the plan is to let the members manage the project(?). He also wrote in the GH issue you linked that `It will be fixed soon.`, maybe he'll be working on porting the project to a new language? All of that is just pure speculation though since I have exactly as much knowledge about what's going on right now as you do.

Maybe @cofyc (maintainer of [dnscrypt-wrapper](https://github.com/cofyc/dnscrypt-wrapper/)) has any idea about the plans for DNSCrypt?

**Comment by Fusl** [comment]
> Would anyone consider that a conflict of interest since I am running the dnscrypt.ca resolvers?
Not really, although it's worth mentioning that the [OpenNIC Project](https://www.opennic.org/) is one of the larger groups using DNSCrypt and as such I believe it should be considered that OpenNIC as a group should maintain the project itself (not the code) as part of the project due to the interest shown.

**Comment by snorkasaurus** [comment]
> Ninja edit: Note: jedisct1 seems to be at the 34C3 event and new year is around the corner so his communication might be limited so we should not speculate too much and allow him a few days to clarify things.

Definitely... I imagine a lot of people are pretty busy at this time of year.  That's one of the reasons your immediate reply kind of caught me off guard. :-)

I'll admit I was a little disappointed that the main project was archived, because it was a central point of contact for providers, maintainers, and users alike.  If you feel it would be more appropriate to use the new DNSCrypt Team area (or elsewhere) to discuss this I am all ears.  My main concern is simply that the project is able to continue.

**Comment by cofyc** [comment]
I don't know much either. @jedisct1 recently created [DNSCrypt team](https://github.com/DNSCrypt), and created lots of new repos. I think he have plans about the future of the DNSCrypt project.

**Comment by Fusl** [comment]
The dnscrypt-proxy repo was now deleted, it is unclear as to why this has happened. Does anyone have an up-to-date fork of it and willing to send over a .tar.gz of the project directory or link me to it? I can try to upload it as new repo to the DNSCrypt team/org and see if it can stay there as archive for now.

**Comment by snorkasaurus** [comment]
Hi Fusl, thanks for the reply!

I actually updated this fork of dnscrypt-proxy on the day that I found out it was set to read-only.  If you are able to use my copy please do of course, and thanks for helping... much appreciated!

**Comment by Fusl** [comment]
Got it upped to https://github.com/DNSCrypt/dnscrypt-proxy, let's see if it stays there. Funny enough, I'm currently the only person "only" having member access in the team while all others have owner access. But that's not going to stop us from resurrecting the project, right?

**Comment by snorkasaurus** [comment]
Additionally I should mention that I just noticed:
  * my repo now says it is a fork of ginhom/dnscrypt-proxy rather than jedisct1/dnscrypt-proxy
  * the dnscrypt.org web site now redirects to https://dnsprivacy.org/wiki/
  * and the [whois for dnscrypt.org](https://www.whois.com/whois/dnscrypt.org) shows a PendingTransfer status

**Comment by snorkasaurus** [comment]
Odd, it seems the dnsprivacy.org web site links to [this tenta.com post](https://tenta.com/blog/post/2017/12/dns-over-tls-vs-dnscrypt) which clearly states DNSCrypt is a dead end (see conclusion section of post).  I'd say that sounds rather ominous. :-(

**Comment by Fusl** [comment]
> my repo now says it is a fork of ginhom/dnscrypt-proxy rather than jedisct1/dnscrypt-proxy

[Mine](https://github.com/Fusl/dnscrypt-proxy) shows the same but I'm pretty sure that's just GitHub algorithm that had to decide which fork is the "base" fork (probably $RANDOM).

**Comment by poentodewo** [comment]
@Fusl https://transfer.sh/kdA9S/LATEST.tar.bz2 maybe can help u version 1.9.5

**Comment by snorkasaurus** [comment]
@Fusl Do you think it might be appropriate to release a v1.9.6 that has a working script/link for the public resolver list? I assume that it would make sense to point the downloads to [here](https://github.com/DNSCrypt/dnscrypt-resolvers/tree/master/v1)?

I also found [this reddit post](https://www.reddit.com/r/privacy/comments/7oiu9f/what_happened_to_dnscrypt/) about the loss of the DNSCrypt repo, and noticed that one of the comments says [Simple DNSCrypt](https://simplednscrypt.org/) is unable to update as well. I actually tried to contact the owners of simplednscrypt.org a while back to tell them their TLSA record was invalid but it was never updated and I never heard back. So I don't know if they are active enough to change their application to point to a new public resolver list anyways (or if they would be willing to use this one).

PS: I also noticed that when I look at the [DNSCrypt profile](https://github.com/DNSCrypt), it looks like you are the only "person" listed. Should there be others?

**Comment by snorkasaurus** [comment]
It appears as though a new project has arrived at [jedisct1/dnscrypt-proxy](https://github.com/jedisct1/dnscrypt-proxy).  There does not seem to be an update-able public resolver list, just a static list in the config file.

A bit of a dick move to fuck around people using the software with no explanation along the way.

**Comment by Fusl** [comment]
I'm slightly worried about the new dnscrypt-proxy version though, especially because issues are disabled for the repository and I haven't seen anything resembling a statement as to why dnscrypt.org is offline and why the repo was taken offline without notice.

**Comment by snorkasaurus** [comment]
I had actually noticed that issues were disabled, and I am still pretty skeptical of it as well.  There is plenty about the last few weeks that doesn't make sense at all.  Also a bit odd is that dnscrypt.org currently resolves to:

  * 95.183.53.20 (solarcom.ch)
  * 107.161.23.204 (ramnode.com)
  * 209.141.38.71 (frantech.ca)

**Comment by Fusl** [comment]
[What. The. Fuck?](http://xor.meo.ws/068bec01/41bd/4d88/b7a7/537d48a77bbc.png)
![](http://xor.meo.ws/068bec01/41bd/4d88/b7a7/537d48a77bbc.png)

**Comment by snorkasaurus** [comment]
Okay I am a little slow... that EUR$18 was sent from Sedo to you?

**Comment by Fusl** [comment]
Yes, with the only comment shown in the screenshot. I'm very confused as to why and who.

**Comment by snorkasaurus** [comment]
Weird... I kind of wish people would start sending me money with strange messages. :-)

**Comment by Fusl** [comment]
Just wanted to update you all that jedisct1 is doing well, he sent me an email explaining what's going on and I understood his concerns. He's currently working on v2 of DNSCrypt and has deleted v1 for various reasons which I can't explain in detail right now, but he is definitely alive, hasn't been hacked and did not get an offer to work at the NSA. I'll be staying in contact with him to figure out how we can get everything rolled over to v2 and new everything without causing too many issues for users.
